mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -512,6 +513,92 @@ object Vectors {
     squaredDistance
   }
 
+  private def dot(a : DenseVector, b : DenseVector) : Double = {
+    (a.toArray zip b.toArray).map(x => (x._1 * x._2)).sum
+  }
+
+  private def dot(a : SparseVector, b : DenseVector) : Double = {
+    (a.indices zip a.values).map(x => { b(x._1)*x._2 }).sum
```

Comment:
What I mean is that you can already just call Breeze for this without any code change in Spark, using `toBreeze`. It's easy; I get that it sometimes involves a copy. Do we have a compelling case for needing this enough and needing it to be fast in Spark that it should be added to `Vector`? I ask just because I don't see anything changed to call it.